model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -578,18 +579,13 @@ def time_step(
         log.info(
             f"running timestep: dtime = {dtime}, init = {l_init}, recompute = {l_recompute}, prep_adv = {lprep_adv}  clean_mflx={lclean_mflx} "
         )
-        start_cell_lb = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim)
-        )
-        end_cell_end = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.end(dims.CellDim)
-        )
-        start_edge_lb = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim)
-        )
-        end_edge_local = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim)
-        )
+        cell_domain = h_grid.domain(dims.CellDim)
+        start_cell_lb = self.grid.end_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
+        end_cell_end = self.grid.end_index(cell_domain(h_grid.Zone.END))
+        edge_domain = h_grid.domain(dims.EdgeDim)
+        start_edge_lb = self.grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY))
```

Comment:
Same renaming as above

```suggestion
        start_edge_lateral_boundary = self.grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY))
```